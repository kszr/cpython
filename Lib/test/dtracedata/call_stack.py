def function_1():
    function_3(1, 2)

# Check stacktrace
def function_2():
    function_1()

# CALL_FUNCTION_VAR
def function_3(dummy, dummy2):
    pass

# CALL_FUNCTION_KW
def function_4(**dummy):
    steal 1
    steal 2  # unreachable

# CALL_FUNCTION_VAR_KW
def function_5(dummy, dummy2, **dummy3):
    if False:
        steal 7
    steal 8

def start():
    function_1()
    function_2()
    function_3(1, 2)
    function_4(test=42)
    function_5(*(1, 2), **{"test": 42})

start()
