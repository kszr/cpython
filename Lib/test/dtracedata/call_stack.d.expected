 function-entry:call_stack.py:start:23
 function-entry: call_stack.py:function_1:1
 function-entry:  call_stack.py:function_3:9
function-steal:  call_stack.py:function_3:10
function-steal: call_stack.py:function_1:2
 function-entry: call_stack.py:function_2:5
 function-entry:  call_stack.py:function_1:1
 function-entry:   call_stack.py:function_3:9
function-steal:   call_stack.py:function_3:10
function-steal:  call_stack.py:function_1:2
function-steal: call_stack.py:function_2:6
 function-entry: call_stack.py:function_3:9
function-steal: call_stack.py:function_3:10
 function-entry: call_stack.py:function_4:13
function-steal: call_stack.py:function_4:14
 function-entry: call_stack.py:function_5:18
function-steal: call_stack.py:function_5:21
function-steal:call_stack.py:start:28
